doc/source/glossary.rst

```diff
@@ -2,6 +2,533 @@
+      operator. In ndarrays, slicing can be applied to every
+      axis:
+
+          >>> a = np.arange(24).reshape(2,3,4)
+          >>> a
+          array([[[ 0,  1,  2,  3],
+                  [ 4,  5,  6,  7],
+                  [ 8,  9, 10, 11]],
+          <BLANKLINE>
+                 [[12, 13, 14, 15],
+                  [16, 17, 18, 19],
+                  [20, 21, 22, 23]]])
+          <BLANKLINE>
+          >>> a[1:,-2:,:-1]
+          array([[[16, 17, 18],
+                  [20, 21, 22]]])
+
+      Trailing slices can be omitted: ::
+
+          >>> a[1] == a[1,:,:]
+          array([[ True,  True,  True,  True],
+                 [ True,  True,  True,  True],
+                 [ True,  True,  True,  True]])
+
+      In contrast to Python, where slicing creates a copy, in NumPy slicing
+      creates a :term:`view`.
+
+      For details, see :ref:`Combining advanced and basic indexing <combining-advanced-and-basic-indexing>`.
+
+
+    ``<``
+
+      In a dtype declaration, indicates that the data is
+      `little-endian <https://en.wikipedia.org/wiki/Endianness>`_
+      (the bracket is big on the right). ::
+
+        >>> dt = np.dtype('<f') # little-endian single-precision float
+
+
+    ``>``
+
+      In a dtype declaration, indicates that the data is
+      `big-endian <https://en.wikipedia.org/wiki/Endianness>`_
+      (the bracket is big on the left). ::
+
+        >>> dt = np.dtype('>H') # big-endian unsigned short
+
+
+    advanced indexing
+
+      Rather than using a :doc:`scalar <reference/arrays.scalars>` or slice as
+      an index, an axis can be indexed with an array, providing fine-grained
+      selection. This is known as :ref:`advanced indexing<advanced-indexing>`
+      or ``fancy indexing``.
+
+
+    along an axis
+
+      The result of an operation along an :term:`axis` X is an array in which X
+      disappears. This can surprise new users expecting the opposite.
+
+      The operation can be visualized this way:
+
+      Imagine a slice of array ``a`` where axis X has a fixed index
+      and the other dimensions are left full (``:``).
+
+          >>> a = np.arange(24).reshape(2,3,4)
+          <BLANKLINE>
+          >>> a.shape
+          (2,3,4)
+          <BLANKLINE>
+          >>> a[:,0,:].shape
+          (2,4)
+
+      The slice has ``a``'s shape with the X dimension deleted. Saying an
+      operation ``op`` is ``performed along X`` means that ``op`` takes as its
+      operands slices having every value of X:
+
+          >>> np.sum(a,axis=1) == a[:,0,:] + a[:,1,:] + a[:,2,:]
+          array([[ True,  True,  True,  True],
+                 [ True,  True,  True,  True]])
+
+
+    array
+
+      Used synonymously in the NumPy docs with
+      :doc:`ndarray <reference/arrays>`, NumPy's basic structure.
+
+
+    array_like
+
+      Any :doc:`scalar <reference/arrays.scalars>` or
+      `sequence <https://docs.python.org/3/glossary.html#term-sequence>`_
```

Comment:
Can be done in a follow-up, but would be nice to replace these with intersphinx links.

Reply:
I wondered if there weren't a better way to handle these. I'm glad to fix it right away. What should it say?

Reply:
My guess would be ``:term:`python:sequence` ``, but that's only a guess.

Reply:
This is a working link, because we have an intersphinx link in `conf.py`:
```
      :ref:`slice <python:glossary>`
```
but we want to link to the `slice` entry, not just the page. This gives "undefined label":
```
      :ref:`slice <python:glossary#term-slice>`
```
as does this:
```
      :ref:`slice <python:glossary:term:slice>`
```
When it comes to half-assed, Sphinx is without peer. 

Leaving the URLs for now.

Reply:
Did my approach fail?

This seems to suggest it should work: https://stackoverflow.com/q/46473481/102441

Reply:
I went ahead and swapped all the explicit links using intersphinx in c24bd and 690037.

I'm against duplicating the terms from the Python glossary in NumPy's glossary, as it just leads to more clicks for the user. Nevertheless I switched them to intersphinx links in 6900372 and leave the decision whether to include them to others.

@bjnath a helpful resource for working with intersphinx is to inspect the objects.inv file, which describes the mapping from the sphinx names to URLs. It's not the most visible thing in the world, but [the intersphinx docs](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#showing-all-links-of-an-intersphinx-mapping-file) explain how to grab `objects.inv` for inspection.

Reply:
> Did my approach fail?

The approach works as expected! I went ahead and applied it throughout.

Reply:
Think I stole the half-assed prize. Sorry @eric-wieser, had the answer right in front of me. Thanks @rossbar for the fixes.

Reply:
Also @rossbar thanks for the `objects.inv` tip.

Reply:
I'll write a guide to linking for NumPy docs writers, including stuff like neps and of course this one.